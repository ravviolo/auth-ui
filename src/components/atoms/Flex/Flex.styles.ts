import styled from 'styled-components';

interface Props {
  direction: 'column' | 'row';
}

export const FlexContainer = styled.div<Props>`
  display: flex;
  flex-direction: ${({ direction }) => direction};
`;
