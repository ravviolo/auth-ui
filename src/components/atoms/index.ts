export { Button } from './Button/Button';
export { Checkbox } from './Checkbox/Checkbox';
export { Divider } from './Divider/Divider';
export { Heading } from './Heading/Heading';
export { IconButton } from './IconButton/IconButton';
export { InputField } from './InputField/InputField';
export { InputLabel } from './InputLabel/InputLabel';
export { TextButton } from './TextButton/TextButton';
