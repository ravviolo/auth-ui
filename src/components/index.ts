export * from './templates/LoginTemplate/LoginTemplate';
export * from './templates/SignUpTemplate/SignUpTemplate';
