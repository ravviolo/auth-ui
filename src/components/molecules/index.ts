export { AuthFooter } from './AuthFooter/AuthFooter';
export { AuthHeader } from './AuthHeader/AuthHeader';
export { AuthSocials } from './AuthSocials/AuthSocials';
export { EmailInput } from './EmailInput/EmailInput';
export { PasswordInput } from './PasswordInput/PasswordInput';
