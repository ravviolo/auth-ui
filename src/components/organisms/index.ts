export * from './LoginForm/LoginForm';
export * from './SignUpForm/SignUpForm';
