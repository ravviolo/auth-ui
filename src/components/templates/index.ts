export * from './LoginTemplate/LoginTemplate';
export * from './SignUpTemplate/SignUpTemplate';
