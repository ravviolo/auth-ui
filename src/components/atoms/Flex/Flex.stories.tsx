import { ComponentMeta, ComponentStory } from '@storybook/react';

import { Flex as FlexComponent } from 'components/atoms';

export default {
  title: 'Atoms/Flex',
  component: FlexComponent,
  argTypes: {
    direction: { control: 'radio' },
  },
  args: {
    children: Array(10)
      .fill(1)
      .map((_, i) => (
        <div
          // eslint-disable-next-line react/no-array-index-key
          key={i}
          style={{
            width: '5rem',
            height: '5rem',
            margin: '1rem',
            backgroundColor: 'cyan',
          }}
        />
      )),
  },
} as ComponentMeta<typeof FlexComponent>;

export const Flex: ComponentStory<typeof FlexComponent> = (args) => (
  <FlexComponent {...args} />
);
