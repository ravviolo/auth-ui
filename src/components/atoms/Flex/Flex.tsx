import * as S from './Flex.styles';

interface Props {
  children: React.ReactNode;
  direction: 'column' | 'row';
  className?: string;
}

export const Flex = ({ direction, className, children }: Props) => {
  return (
    <S.FlexContainer className={className} direction={direction}>
      {children}
    </S.FlexContainer>
  );
};
