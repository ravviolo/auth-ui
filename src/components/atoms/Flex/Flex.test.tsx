import { render, screen, within } from '@testing-library/react';
import { create } from 'react-test-renderer';

import { Flex } from 'components/atoms';

import { Props as FlexProps } from './Flex';

describe('Flex', () => {
  const props: Omit<FlexProps, 'direction'> = {
    testId: 'test-flex',
    children: Array(10)
      .fill(1)
      .map((_, i) => (
        <li
          // eslint-disable-next-line react/no-array-index-key
          key={i}
        />
      )),
  };

  it('renders all children horizontally', () => {
    render(<Flex {...props} direction="row" />);

    const flexContainer = screen.getByTestId('test-flex');

    const flexChildren = within(flexContainer).getAllByRole('listitem');
    expect(flexChildren).toHaveLength(10);

    const flexContainerStyles = window.getComputedStyle(flexContainer);
    expect(flexContainerStyles.flexDirection).toBe('row');
  });

  it('renders all children vertically', () => {
    render(<Flex {...props} direction="column" />);

    const flexContainer = screen.getByTestId('test-flex');

    const flexChildren = within(flexContainer).getAllByRole('listitem');
    expect(flexChildren).toHaveLength(10);

    const flexContainerStyles = window.getComputedStyle(flexContainer);
    expect(flexContainerStyles.flexDirection).toBe('column');
  });

  it('matches snapshot - renders all children horizontally', () => {
    const tree = create(<Flex {...props} direction="row" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot - renders all children vertically', () => {
    const tree = create(<Flex {...props} direction="column" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
