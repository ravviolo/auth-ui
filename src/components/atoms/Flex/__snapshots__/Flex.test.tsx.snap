// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex should match snapshot - renders all children horizontally 1`] = `
<div
  className="sc-jSMfEi iChCMC"
  data-testid="test-flex"
  direction="row"
>
  <li />
  <li />
  <li />
  <li />
  <li />
  <li />
  <li />
  <li />
  <li />
  <li />
</div>
`;

exports[`Flex should match snapshot - renders all children vertically 1`] = `
<div
  className="sc-jSMfEi bQrDqy"
  data-testid="test-flex"
  direction="column"
>
  <li />
  <li />
  <li />
  <li />
  <li />
  <li />
  <li />
  <li />
  <li />
  <li />
</div>
`;
